File: exercism/typescript/word-count/word-count.ts
export function count(phrase: string): Map<string | number, number> {
  let result: Map<string | number, number> = new Map<string | number, number>()

  let words: string[] = phrase.trim().split(new RegExp("[ \\t\\n][ ]*"));
  let wordsLower: string[] = words.map(x => x.toLowerCase());

  for (let word of wordsLower) {
    if (result.has(word)) {
      // @ts-ignore
      result.set(word, result.get(word) + 1);
    } else {
      result.set(word, 1);
    }
  }

  return result;
}

File: exercism/python/word-count/word_count.py
def count_words(sentence: str) -> {}:
    sentence = sentence.lower() \
        .replace(",", " ") \
        .replace(".", " ") \
        .replace(":", " ") \
        .replace("!", " ") \
        .replace("&", " ") \
        .replace("@", " ") \
        .replace("$", " ") \
        .replace("%", " ") \
        .replace("^", " ") \
        .replace("_", " ") \
        .replace("\n", " ")

    words_in_sentence = sentence.split()

    words = dict()
    for word in words_in_sentence:
        while (word.find("'") != -1):
            if word[0] == "'":
                word = word[1::]
            elif word[-1] == "'":
                word = word[:-1:]
            else:
                break

        if word.strip() == "":
            continue

        if word not in words:
            words[word] = 1
        else:
            words[word] += 1

    return words
